sdk/core/azure-core/azure/core/async_paging/_async_page_iterator.py

```diff
@@ -0,0 +1,107 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+import collections.abc
+import logging
+from typing import (
+    AsyncIterator,
+    Awaitable,
+    Callable,
+    Iterable,
+    Optional,
+    Tuple,
+)
+
+from ._async_paging_method_handler import _AsyncPagingMethodHandler
+from ._async_list import AsyncList
+from ..paging._utils import ReturnType, ResponseType
+from ..paging import PagingMethodABC
+from ..exceptions import AzureError
+from ..pipeline._tools_async import await_result as _await_result
+
+_LOGGER = logging.getLogger(__name__)
+
+class AsyncPageIterator(AsyncIterator[AsyncIterator[ReturnType]]):
+    def __init__(
+        self,
+        get_next: Optional[Callable[
+            [Optional[str]], Awaitable[ResponseType]
+        ]] = None,
+        extract_data: Optional[Callable[
+            [ResponseType], Awaitable[Tuple[str, AsyncIterator[ReturnType]]]
+        ]] = None,
+        continuation_token: Optional[str] = None,
+        paging_method: Optional[PagingMethodABC] = None,
```

Comment:
We should make at least `paging_method` a keyword-only argument for the AsyncPageIterator. Per our guidelines, all optional arguments should have been keyword-only, but we shouldn't make a breaking change just for that.

Reply:
made them both kwargs for good measure